packages/modules/influxdb/src/influxdb-container.ts

```diff
@@ -0,0 +1,242 @@
+    this.httpWait.withBasicCredentials(this.username, this.password);
+    return this;
+  }
+
+  public withPassword(password: string): this {
+    this.password = password;
+    // keep wait strategy credentials up to date (mainly relevant for 1.x)
+    this.httpWait.withBasicCredentials(this.username, this.password);
+    return this;
+  }
+
+  public withOrganization(organization: string): this {
+    this.organization = organization;
+    return this;
+  }
+
+  public withBucket(bucket: string): this {
+    this.bucket = bucket;
+    return this;
+  }
+
+  public withRetention(retention: string): this {
+    this.retention = retention;
+    return this;
+  }
+
+  public withAdminToken(adminToken: string): this {
+    this.adminToken = adminToken;
+    return this;
+  }
+
+  // InfluxDB 1.x specific methods
+  public withAuthEnabled(authEnabled: boolean): this {
+    this.authEnabled = authEnabled;
+    return this;
+  }
+
+  public withAdmin(admin: string): this {
+    this.admin = admin;
+    return this;
+  }
+
+  public withAdminPassword(adminPassword: string): this {
+    this.adminPassword = adminPassword;
+    return this;
+  }
+
+  public withDatabase(database: string): this {
+    this.database = database;
+    return this;
+  }
+
+  public override async start(): Promise<StartedInfluxDBContainer> {
+    // Re-evaluate version based on the final image tag
+    this.isVersion2 = this.isInfluxDB2(this.imageName.tag ?? "latest");
+    if (this.isVersion2) {
+      this.configureInfluxDB2();
+    } else {
+      this.configureInfluxDB1();
+    }
+
+    return new StartedInfluxDBContainer(
+      await super.start(),
+      this.username,
+      this.password,
+      this.organization,
+      this.bucket,
+      this.database,
+      this.adminToken,
+      this.isVersion2
+    );
+  }
+
+  private configureInfluxDB2(): void {
+    const env: Record<string, string> = {
+      DOCKER_INFLUXDB_INIT_MODE: "setup",
+      DOCKER_INFLUXDB_INIT_USERNAME: this.username,
+      DOCKER_INFLUXDB_INIT_PASSWORD: this.password,
+      DOCKER_INFLUXDB_INIT_ORG: this.organization,
+      DOCKER_INFLUXDB_INIT_BUCKET: this.bucket,
+    };
+
+    if (this.retention) {
+      env.DOCKER_INFLUXDB_INIT_RETENTION = this.retention;
+    }
+
+    if (this.adminToken) {
+      env.DOCKER_INFLUXDB_INIT_ADMIN_TOKEN = this.adminToken;
+    }
+
+    this.withEnvironment(env);
+  }
+
+  private configureInfluxDB1(): void {
+    const env: Record<string, string> = {
+      INFLUXDB_USER: this.username,
+      INFLUXDB_USER_PASSWORD: this.password,
+      INFLUXDB_HTTP_AUTH_ENABLED: String(this.authEnabled),
+      INFLUXDB_ADMIN_USER: this.admin,
+      INFLUXDB_ADMIN_PASSWORD: this.adminPassword,
+    };
+
+    if (this.database) {
+      env.INFLUXDB_DB = this.database;
+    }
+
+    this.withEnvironment(env);
+  }
+
+  private isInfluxDB2(tag: string): boolean {
+    if (tag === "latest") {
+      return true; // Assume latest is v2
```

Comment:
Does this module work for InfluxDB v3? If so I'd change the naming of this logic to be either `v1` or `> v1`, instead of `v1` or `v2` or `...`